lib/rules/logical-assignment-operators.js

```diff
@@ -0,0 +1,430 @@
+    if (node.type === "Program") {
+        return false;
+    }
+
+    return node.parent.type === "WithStatement" && node.parent.body === node ? true : isInsideWithBlock(node.parent);
+}
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+/** @type {import('../shared/types').Rule} */
+module.exports = {
+    meta: {
+        type: "suggestion",
+
+        docs: {
+            description: "Require or disallow assignment logical operator shorthand",
+            recommended: false,
+            url: "https://eslint.org/docs/rules/logical-assignment-operators"
+        },
+
+        schema: {
+            type: "array",
+            oneOf: [{
+                items: [
+                    { const: "always" },
+                    {
+                        type: "object",
+                        properties: {
+                            enforceForIfStatements: {
+                                type: "boolean"
+                            }
+                        },
+                        additionalProperties: false
+                    }
+                ],
+                minItems: 0, // 0 for allowing passing no options
+                maxItems: 2
+            }, {
+                items: [{ const: "never" }],
+                minItems: 1,
+                maxItems: 1
+            }]
+        },
+        fixable: "code",
+        // eslint-disable-next-line eslint-plugin/require-meta-has-suggestions -- Does not detect conditional suggestions
+        hasSuggestions: true,
+        messages: {
+            assignment: "Assignment (=) can be replaced with operator assignment ({{operator}}).",
+            useLogicalOperator: "Convert this assignment to use the operator {{ operator }}.",
+            logical: "Logical expression can be replaced with an assignment ({{ operator }}).",
+            if: "If can be replaced with operator assignment.",
+            convertIf: "Replace this if with an logical assignment.",
+            unexpected: "Unexpected logical operator assignment ({{operator}}) shorthand.",
+            separate: "Separate the logical assignment into an assignment with a logical operator."
+        }
+    },
+
+    create(context) {
+        const mode = context.options[0] === "never" ? "never" : "always";
+        const checkIf = mode === "always" && context.options.length > 1 && context.options[1].enforceForIfStatements;
+        const sourceCode = context.getSourceCode();
+
+        /**
+         * Do not fix if the access could be a getter
+         * @param {ASTNode} assignment Assignment expression
+         * @returns {boolean} True iff the fix is safe
+         */
+        function shouldBeFixed(assignment) {
+            return assignment.left.type === "Identifier" &&
+                   (context.getScope().isStrict || !isInsideWithBlock(assignment));
```

Comment:
Why do we have `context.getScope().isStrict` in this condition?

Reply:
This is purely for speedup. There can be no `with` block inside a strict scope.

Reply:
True, but that strict scope could be inside `with`:

```js
with (a) {
  (function() {
      "use strict";
      b = b || c; // this could be a.b = a.b || c
  })();
}
```

Reply:
You are right. I changed it to use to check whether the global scope is strict.